### PR TITLE
Filter invalid line items report

### PR DESCRIPTION
**Pull request by rich-nguyen**
The invalid line items report would be clearer if:

- line items which exclusive target other gu sites using 'nearly' top level ad units, like `jobs.theguardian.com` and `apps`, were understood and grouped.

- automated line items from sonobi were grouped, leaving true unknowns left in the list.

I'd like to do both, but this change sorts out the first, and it should be quite easy to update the `groupBy` in future to understand `jobs.thguardian.com`. @ScottPainterGNM 

screen shot 2017-01-05 at 10 59 35

### CODE REVIEW

**Comment by rich-nguyen** [comment]
[Improve clarity of invalid line items report page](https://trello.com/c/d45S3gkK/53-improve-clarity-of-invalid-line-items-report-page)

**Comment by rich-nguyen** [comment]
retro plus one needed, this is for a demo (now).

File: admin/app/controllers/admin/CommercialController.scala [hunk 1]
```diff
@@ -114,21 +114,34 @@ class CommercialController(implicit context: ApplicationContext) extends Control
   def renderInvalidItems() = Action { implicit request =>
     // If the invalid line items are run through the normal extractor, we can see if any of these
     // line items appear to be targeting Frontend.
-    val invalidLineItems = Store.getDfpLineItemsReport().invalidLineItems
+    val invalidLineItems: Seq[GuLineItem] = Store.getDfpLineItemsReport().invalidLineItems
     val invalidItemsExtractor = DfpDataExtractor(invalidLineItems, Nil)
 
+    val advertisers = AdvertiserAgent.get
+    val orders = OrderAgent.get
+    val sonobiAdvertiserId = advertisers.find(_.name.toLowerCase =="sonobi").map(_.id).getOrElse(0L)
```
Comment (kelvin-chappell) [inline admin/app/controllers/admin/CommercialController.scala]: If no sonobi advertiser ID, perhaps don't show any sonobi results?

File: admin/app/dfp/OrderAgent.scala [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+object OrderAgent {
+
+  private lazy val cache = AkkaAgent(Seq.empty[GuOrder])
+
+  def refresh(): Future[Seq[GuOrder]] = {
+    cache alterOff { oldData =>
+      val freshData = DfpApi.getAllOrders
+      if (freshData.nonEmpty) freshData else oldData
+    }
+  }
```
Comment (kelvin-chappell) [inline admin/app/dfp/OrderAgent.scala]: I wonder if this refresh pattern can be encapsulated for different types, as it's repeated elsewhere.

**Review by kelvin-chappell — COMMENTED** [review]
Looks good.  Made some suggestions.
🚢  
Too late!

File: admin/app/controllers/admin/CommercialController.scala (hunk 1 above)
Comment (rich-nguyen) [inline admin/app/controllers/admin/CommercialController.scala]: if there wasn't a sonobi advertiser, the sonobi section of the report would display _None_

File: admin/app/dfp/OrderAgent.scala (hunk 2 above)
Comment (rich-nguyen) [inline admin/app/dfp/OrderAgent.scala]: Yeah, certainly could be tidier. Also think the folder organisation could be neater to find all the agents easily.

**Pull request merged**